From 0c9e8bbbc3d34d11b2c2e4c6f3dfbb3a4f1b16c7 Mon Sep 17 00:00:00 2001
From: Jinzhe Zeng <jinzhe.zeng@rutgers.edu>
Date: Thu, 30 Nov 2023 17:53:07 -0500
Subject: [PATCH 2/2] Compile using C++17 when using PyTorch 2.1.0

---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 693e03ada..5815671c9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -372,8 +372,12 @@ if (ENABLE_FLOAT_FFTW)
 endif()
 
 if(ENABLE_DEEPKS)
-  set_if_higher(CMAKE_CXX_STANDARD 14)
   find_package(Torch REQUIRED)
+  if(NOT Torch_VERSION VERSION_LESS "2.1.0")
+    set_if_higher(CMAKE_CXX_STANDARD 17)
+  elseif(NOT Torch_VERSION VERSION_LESS "1.5.0")
+    set_if_higher(CMAKE_CXX_STANDARD 14)
+  endif()
   include_directories(${TORCH_INCLUDE_DIRS})
   target_link_libraries(${ABACUS_BIN_NAME} deepks)
   list(APPEND math_libs ${TORCH_LIBRARIES})
-- 
2.34.1

